### PR TITLE
AC-5775: add office hours availability to new profile page

### PR DESCRIPTION
**Pull request by shanbady**
This PR adds an "officehoursUrl" attribute to the graphql response as part of https://github.com/masschallenge/directory/pull/84

### CODE REVIEW

File: web/impact/impact/graphql/types/expert_profile_type.py [hunk 1]
```diff
@@ -51,13 +53,28 @@ def resolve_available_office_hours(self, info, **kwargs):
         user = info.context.user
         filter_kwargs = {
             'finalist__isnull': True,
-            'date__gte': timezone.now(),
-            'start_time__gte': timezone.now().time()
         }
+        now = timezone.now()
```
Comment (yotammanor) [inline web/impact/impact/graphql/types/expert_profile_type.py]: something I just noticed - a user who sees their own profile page should also see available mentor office hours (currently it isn't the case). and also superusers, probably.

**Review by yotammanor — CHANGES_REQUESTED** [review]
a few small requests, mostly good.

File: web/impact/impact/graphql/types/expert_profile_type.py [hunk 2]
```diff
@@ -51,13 +53,28 @@ def resolve_available_office_hours(self, info, **kwargs):
...
+              start_time__gte=now.time()) | Q(
+                date__gt=now)).exists()
+
+    def resolve_office_hours_url(self, info, **kwargs):
+        if self.user.programrolegrant_set.filter(
+                program_role__user_role__name=UserRole.MENTOR
+        ).exists():
+            latest_grant = self.user.programrolegrant_set.filter(
+                program_role__user_role__name=UserRole.MENTOR
+            ).latest('created_at')
```
[the remark this answers is not in the document]
Comment (shanbady) [reply web/impact/impact/graphql/types/expert_profile_type.py:72]: i previously had it this way but it threw a 'program role grant does not exist' exception

File: web/impact/impact/graphql/types/expert_profile_type.py [hunk 3]
```diff
@@ -51,13 +53,28 @@ def resolve_available_office_hours(self, info, **kwargs):
...
         }
+        now = timezone.now()
         if not user.is_staff:
             filter_kwargs['program__in'] = _get_user_programs(user)
-        return self.user.mentor_officehours.filter(
-            **filter_kwargs).exists()
+        return self.user.mentor_officehours.filter(**filter_kwargs).filter(
+            Q(date=now,
+              start_time__gte=now.time()) | Q(
+                date__gt=now)).exists()
```
[the remark this answers is not in the document]
Comment (shanbady) [reply web/impact/impact/graphql/types/expert_profile_type.py]: done

File: web/impact/impact/__init__.py [hunk 4]
```diff
@@ -1,2 +1,2 @@
 
-__version__ = '1.0.0'
\ No newline at end of file
+__version__ = '1.0.1'
```
[the remark this answers is not in the document]
Comment (shanbady) [reply web/impact/impact/__init__.py]: reverted

File: web/impact/impact/graphql/types/expert_profile_type.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (pbx) [reply web/impact/impact/graphql/types/expert_profile_type.py]: seems like there's a middle ground there though. 
```
mentor_grant_qs = self.user.programrolegrant_set.filter(
                program_role__user_role__name=UserRole.MENTOR
            )
if mentor_grant.qs.exists():
    latest_grant = mentor_grant_qs.latest()
    # do stuff
```

[the remark this answers is not in the document]
Comment (pbx) [reply web/impact/impact/graphql/types/expert_profile_type.py:72]: Seems like there's a middle ground: 
```python
mentor_grants = self.user.programrolegrant_set.filter(
                program_role__user_role__name=UserRole.MENTOR
            )
if mentor_grants.exists():
    latest_grant = mentor_grants.latest('created_at')
    # do stuff
```

**Review by yotammanor — APPROVED** [review]
Thanks for picking up the changes!

**Pull request merged**